sandbox/samples/cpp/alembicassembly/alembicassembly.cpp

```diff
@@ -1163,6 +1164,86 @@ namespace
             return Model;
         }
 
+        // Return metadata for this assembly model.
+        asf::Dictionary get_model_metadata() const override
+        {
+            return
+                asf::Dictionary()
+                    .insert("name", Model)
+                    .insert("label", "Alembic Assembly");
+        }
+
+        // Return metadata for the inputs of this object model.
+        asf::DictionaryArray get_input_metadata() const override
+        {
+            asf::DictionaryArray metadata;
+
+            metadata.push_back(
+                asf::Dictionary()
+                    .insert("name", "file_path")
+                    .insert("label", "File path")
+                    .insert("type", "text")
+                    .insert("use", "required"));
+
+            metadata.push_back(
```

Comment:
Are these supposed to be the metadata for the pinhole camera?

Reply:
No, those are assembly parameters use to sample the given alembic file.

For now, if not setted, [it use `0.0f`](https://github.com/appleseedhq/appleseed/blob/01a053de46c70359511dcc2494e59e364edcbfc7/sandbox/samples/cpp/alembicassembly/alembicassembly.cpp#L432).

IIRC there is no way to retrieve the shutter open/close of the current scene, so you need to pass it explicitly to each assembly.

Doc [here](https://github.com/appleseedhq/appleseed/wiki/Alembic-Assembly#shutter_open_time-and-shutter_close_time-float-optional)

The camera doesn't have shutter open/close but  [explicit `time_offset`](https://github.com/appleseedhq/appleseed/wiki/Alembic-Camera#time_offset-float)

Any suggestion ?

Reply:
> IIRC there is no way to retrieve the shutter open/close of the current scene, so you need to pass it explicitly to each assembly.

You _can_ retrieve the shutter times of the camera. In `do_expand_contents()` you've got a `Project` so you first need to retrieve the active camera used for rendering:

    const Camera* camera = project.get_scene()->get_active_camera();

Then you can use `camera->get_shutter_open_begin_time()`, etc. to retrieve shutter times.